### PR TITLE
Update start stop api to use pod status function.

### PR DESCRIPTION
**Pull request by sujil02**
Update the start and stop api to validate pod status based on status functions
rather than iterating over all the containers in the pod. (Addressing the TODO)

Signed-off-by: Sujil02 <sushah@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/libpod/pods.go [hunk 1]
```diff
@@ -172,34 +172,19 @@ func PodStop(w http.ResponseWriter, r *http.Request) {
...
 	pod, err := runtime.LookupPod(name)
 	if err != nil {
 		utils.PodNotFound(w, name, err)
 		return
 	}
 
-	// TODO we need to implement a pod.State/Status in libpod internal so libpod api
-	// users dont have to run through all containers.
-	podContainers, err := pod.AllContainers()
+	status, err := pod.GetPodStatus()
```
Comment (baude) [inline pkg/api/handlers/libpod/pods.go]: remind me, does this also account for the infra container?

Comment (sujil02) [reply pkg/api/handlers/libpod/pods.go]: Yes, it does takes the infra container into account.  pod.GetPodStatus() and pod.AllContainers() both internally call the runtime.state.PodContainers hence no major change in the implementaion

**Comment by rhatdan** [comment]
@sujil02 needs a rebase.

**Comment by mheon** [comment]
/approve
LGTM on my end

**Pull request merged**